libgalois/src/GraphTopology.cpp

```diff
@@ -435,6 +435,198 @@ katana::EdgeTypeAwareTopology::MakeFrom(
       pg, edge_type_index, e_topo, std::move(per_type_adj_indices)});
 }
 
+std::unique_ptr<katana::ProjectedTopology>
+katana::ProjectedTopology::MakeTypeProjectedTopology(
+    const katana::PropertyGraph* pg, const std::vector<std::string>& node_types,
+    const std::vector<std::string>& edge_types) {
+  KATANA_LOG_DEBUG_ASSERT(pg);
+
+  const auto& topology = pg->topology();
+  if (topology.empty()) {
+    ProjectedTopology pst;
+    return std::make_unique<ProjectedTopology>(std::move(pst));
```

Comment:
Why use 2 lines to do what can be done in one line?

Reply:
fixed now